superset/db_engine_specs/drill.py

```diff
@@ -68,3 +68,12 @@ def convert_dttm(cls, target_type: str, dttm: datetime) -> Optional[str]:
     def adjust_database_uri(cls, uri: URL, selected_schema: Optional[str]) -> None:
         if selected_schema:
             uri.database = parse.quote(selected_schema, safe="")
+
+    @classmethod
+    def modify_url_for_impersonation(cls, url: URL, impersonate_user: bool, username: Optional[str]) -> None:
+        if impersonate_user and username is not None:
+            if url.drivername == "drill+odbc":
+                url.query["DelegationUID"] = username
```

Comment:
it looks like this query param applies to odbc and jdbc but not sadrill, is that correct?
cc @betodealmeida

Reply:
For JDBC the parameter would be `impersonation_target` as you can see here --> https://drill.apache.org/docs/configuring-inbound-impersonation/ but we have not tested this.

Should i include a case for `drill+jdbc` with this parameter even if i have not tested this?

Reply:
@Z0ltrix I personally think you can include it if it is expected to work even if you haven't been able to test it yourself; if it doesn't someone else can continue the work by posting a follow-up PR.

Reply:
added impersonation_target for jdbc

Reply:
@villebro , @eschutho ,
can we resolve this conversation?
Regards,
Christian

Reply:
Thanks @Z0ltrix and @villebro. Looks good!